app/models/application_plan.rb

```diff
@@ -17,7 +17,10 @@ class ApplicationPlan < Plan
       where(services: { account_id: provider.id }).joins(:service).references(:service).readonly(false)
     end
   }
-
+  validates :trial_period_days, numericality: { greater_than_or_equal_to: 0 }
+  validates :setup_fee, numericality: { greater_than_or_equal_to: 0.00 }
+  validates :cost_per_month, numericality: { greater_than_or_equal_to: 0.00 }
```

Comment:
Notice that we already have `numericality` validation defined for `cost_per_month` at https://github.com/3scale/porta/blob/179cd1fd95ff5ecee03618311053f2f81fb698cf/app/models/plan.rb#L64.

Perhaps all these should move to `Plan`?

Also, make sure to include `allow_nil: true` option.

Reply:
Done